lightaqonly.lua

```diff
@@ -0,0 +1,32 @@
+-- Changes heavy aquifers to light pre embark
+local help = [====[
+
+lightaqonly
+===========
+Changes the Drainage of all world tiles that would generate Heavy aquifers into
+a value that results in Light aquifers instead.
+
+Note that the script has to be run before embarking to have any effect on an embark.
+
+]====]
+function lightaqonly ()
+  if not dfhack.isWorldLoaded () then
+    qerror ("Error: This script requires a world to be loaded.")
+  end
+
+  if dfhack.isMapLoaded () then
+    qerror ("Error: This script requires a world to be loaded, but not a map.")
+  end
+
+  for i = 0, df.global.world.world_data.world_width - 1 do
+    for k = 0, df.global.world.world_data.world_height - 1 do
+      local tile = df.global.world.world_data.region_map [i]:_displace (k)
+      
+      if tile.drainage % 20 == 7 then
```

Comment:
This is some interesting math - do you know what the values of `drainage` represent? I don't think it could be an enum or a bitfield going by this. Just curious why e.g. changing `7` to `8` and `27` to `28` do the same thing.

Reply:
Yes, Drainage is one of the PSV values, one that has the "normal" range of 0 - 100 (and so, essentially a percentage), which influences what the biome is. The math is straight from Toady's answer to my question in the latest FotF, which essentially is that it's a crude "RNG" to get heavy aquifers about 5 % of the time, basically because there were no spare bits available. I did check that adding 1 did not result in the crossing of a biome boundary for any of the values, and I believe I could have removed one instead.

Reply:
A link to either FOTF or the wiki to clarify where these numbers came from would be great